### DISCUSSION TITLE
Checkerboard Tiles

### DISCUSSION THREAD

**Discussion by SamBigAbs** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#109] [Ideas]
I only have a surface level understanding of how this technology works but would it be possible to do half of the tiles in one pass and the others in the second pass as a checkerboard? I do not mean alternating tiles but  performing 20 steps on the first set before doing the same on the 2nd. The benefit of this would hopefully be better consistency and larger possible images. 
It would be like switching between inpainting masked and unmasked. 
![image](https://user-images.githubusercontent.com/50723098/230451978-0c90b80f-c1bc-4c9d-98fb-f24a79631fe1.png)

**Comment by pkuliyi2015** [comment]
Thank you for paying attention to the technology part

The chessboard a strategy has been applied in ultimate SD upscaler. However, it seems the strategy is not as effective as you think, because the inconsistency between mask no mask areas is too large to be compensated by the chessboard inpainting.

I have found a relatively better way to increase the consistency. This is realized by Tiled noise inversion, which can find the optimal noise for each tile window and greatly increase the consistency. The chessboard inpainting strategy is not needed anymore as the tiles are already very consistent.

If there is no technical advancement recently, this is the optimal solution I think. I’m currently the state-of-the-art.

**Reply by SamBigAbs** [reply]
Thanks for the response!
I have been using this extension a lot the past few weeks with great results. 
I was unaware that was the purpose of the noise inversion. I tried out various configurations but it remains mostly a mystery to me. 
If you want you want help or feedback on features or someone to assist with tutorials feel free to message me.

**Reply by SamBigAbs** [reply]
I think you are correct about the chessboard tiling not being a solution. If you have the time to humour me then I have another suggestion based on my observations. 
The biggest problem is that the details added to the image need the context of the larger image. Let's say you are upscaling a face with glasses from 512 to 2048. At 512 SD recognizes the glasses but when only seeing a quarter of the image it may not have enough context to know what objects are. 
Somehow the upscaler needs to combine the context of smaller resolutions with the detail of higher resolutions. 
Stable Diffusion needs to be able to "see" the entire image for context while working on a portion of it at a high resolution. Completing each step at various tile sizes then averaging them may be a solution.
Another suggestion is to downscale the resulting image each step and compare it to the original and or previous image. If the difference between the images is too large then the denoising strength could be lowered until it is below the threshold.
Those are just some ideas I have. You don't have to explain why I'm wrong if they are stupid.
